packages/drawer/src/views/modern/Drawer.tsx

```diff
@@ -207,11 +207,15 @@ export default function Drawer({
           : open;
 
       toggleDrawer(nextOpen, event.velocityX);
-    },
-    onFinish: () => {
+    })
+    .onFinalize(() => {
       runOnJS(onGestureFinish)();
-    },
-  });
+    });
+  // @ts-expect-error Ref types incompatible yet, needs to get fixed upstream
+  panGesture.config = {
+    ...panGesture.config,
+    ...gestureHandlerProps,
```

Comment:
Hmm, this raises some more problems.

Currently, [`gestureHandlerProps`](https://reactnavigation.org/docs/drawer-navigator/#gesturehandlerprops) allows users to customise `PanGestureHandler` simply by passing props to it (with `React.ComponentProps<typeof PanGestureHandler>` signature).

After some consultation with @j-piasecki we've come to conclusion there's no guarantee whatsoever that props coming from v1 are going to work with RNGH v2. Or maybe the other way around - there are for sure going to be more props in RNGH 2.

We would have to adjust the API on how users can customize gestures in React Navigation v7.

The solution @j-piasecki and I propose is to drop the `gestureHandlerProps` prop in favor of the `panGesture` prop. Users would be able to pass `Gesture.Pan()` with options the user wants to add or override. This then would have to be merged right in this selected line.

Reply:
oh yeah def, that sounds like a clean API. Shall I make that change in the props, or will you do at some point in the `7.x` branch?